src/pyo3_extensions/chrono_tz.rs

```diff
@@ -0,0 +1,232 @@
+use chrono::{
+    DateTime, Datelike, Duration, NaiveDate, NaiveDateTime, NaiveTime, Offset, TimeZone, Timelike,
+};
+use log::warn;
+use pyo3::{
+    exceptions::PyTypeError,
+    types::{PyDateAccess, PyDateTime, PyTimeAccess, PyTzInfoAccess},
+    FromPyObject, IntoPy, PyAny, PyObject, PyResult, PyTryFrom, Python, ToPyObject,
+};
+use serde::{de::SeqAccess, ser::SerializeTupleStruct, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Copy, PartialOrd)]
+pub struct ChronoDateTime(pub(crate) DateTime<chrono_tz::Tz>);
+
+/// Utilities for the wrapper DateTime
+impl ChronoDateTime {
+    pub(crate) fn max_utc() -> Self {
+        Self(chrono_tz::Tz::UTC.from_utc_datetime(&chrono::MAX_DATETIME.naive_utc()))
+    }
+
+    pub(crate) fn max_local() -> Self {
+        Self(chrono_tz::Tz::UTC.from_utc_datetime(&chrono::MAX_DATETIME.naive_local()))
+    }
+
+    pub(crate) fn utc_now() -> Self {
+        Self(chrono_tz::UTC.from_utc_datetime(&chrono::Utc::now().naive_local()))
+    }
+
+    /// Proxy DateTime::checked_add_signed, mapping result to wrap it in Self
+    pub fn checked_add_signed(&self, dur: chrono::Duration) -> Option<Self> {
+        self.0.checked_add_signed(dur).map(Self)
+    }
+}
+
+impl std::ops::Sub<&ChronoDateTime> for &ChronoDateTime {
+    type Output = Duration;
+
+    #[inline]
+    fn sub(self, rhs: &ChronoDateTime) -> Duration {
+        self.0 - rhs.0
+    }
+}
+
+impl std::ops::Sub<ChronoDateTime> for ChronoDateTime {
+    type Output = Duration;
+
+    #[inline]
+    fn sub(self, rhs: ChronoDateTime) -> Duration {
+        self.0 - rhs.0
+    }
+}
+
+impl ToPyObject for ChronoDateTime {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
```

Comment:
Is there a way to piggyback on the PyO3 built-in TZ-ignoring conversions and then fill in the TZ info here, rather than re-creating the rest of the conversion? Or does that make things more confusing?

Reply:
I'll look into it, it makes sense to reuse it if we can